warehouse/models/intermediate/gtfs/gtfs_quality/int_gtfs_quality__shapes_for_all_trips.sql

```diff
@@ -0,0 +1,49 @@
+WITH feed_guideline_index AS (
+    SELECT * FROM {{ ref('int_gtfs_quality__schedule_feed_guideline_index') }}
+),
+
+dim_trips AS (
+    SELECT * FROM {{ ref('dim_trips') }}
+),
+
+summarize_trips AS (
+   SELECT
+       feed_key,
+       EXTRACT(date FROM _valid_from) AS valid_from,
+       EXTRACT(date FROM _valid_to) AS valid_to,
+       COUNTIF(shape_id IS NOT NULL) AS ct_shape_trips,
```

Comment:
I am not sure if you want to do the extra step of checking if that shape ID actually can join with the shapes table, to check if they're all actually valid foreign keys? I suppose there is a separate check for dangling refs so maybe it doesn't matter, just wanted to ask.

Reply:
this is tempting, I think I could argue either way on it

Reply:
Totally defer to you, I don't really think one is clearly "better". 

If the intent is "they've listed a shape for every trip", I think leaving it as-is is probably simplest and has exactly one failure case (they didn't list a shape). Once you do the join you have to think about fanout and issues in the shapes table.

Doing the join would be more complete/comprehensive in terms of "there is an actual valid shape for every trip". But you might have to change the logic to check that there is exactly one match. And then you have multiple failure conditions.